src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -105,37 +96,49 @@ class JooqCalibrateDataRepository: CalibrateDataRepository
         return arrayList
     }
 
-    @Suppress("MagicNumber")
+    @Suppress("UnsafeCallOnNullableType")
     private fun getFilteredDataFromConnection(
         conn: Connection,
-        filterQuery: FilterQuery): List<CalibrateResultRow> {
-        val resultSet: ResultSet
-        var query = FILTER_TEMPLATE
-
-        for ((field, values) in filterQuery) {
-            if (values.size > 0) {
-                val questionMarks = values.map { "?" }.joinToString(",")
-                query += "$field IN ($questionMarks) AND "
-            }
-        }
-        query = query.dropLast(5)
-
-        val stmt: PreparedStatement = conn.prepareStatement(query)
-        var stmtIndex = 1;
-        for ((_, values) in filterQuery) {
-            values.forEach { it ->
-                if (it is String) {
-                    stmt.setString(stmtIndex, it)
-                } else if (it is Int) {
-                    stmt.setInt(stmtIndex, it)
+        filterQuery: FilterQuery): List<CalibrateResultRow>
+    {
+        val dataSource = SingleConnectionDataSource(conn)
+        return Database.connect(dataSource)
+            .from(ResultData)
+            .select()
+            .whereWithConditions {
+                if (filterQuery.indicator.size > 0) {
+                    it += ResultData.indicator inList filterQuery.indicator
+                }
+                if (filterQuery.calendarQuarter.size > 0) {
+                    it += ResultData.calendarQuarter inList filterQuery.calendarQuarter
+                }
+                if (filterQuery.ageGroup.size > 0) {
+                    it += ResultData.ageGroup inList filterQuery.ageGroup
+                }
+                if (filterQuery.sex.size > 0) {
+                    it += ResultData.sex inList filterQuery.sex
+                }
+                if (filterQuery.areaId.size > 0) {
+                    it += ResultData.areaId inList filterQuery.areaId
+                }
+                if (filterQuery.areaLevel.size > 0) {
+                    it += ResultData.areaLevel inList filterQuery.areaLevel
                 }
-                stmtIndex++
             }
-        }
-
-        resultSet = stmt.executeQuery()
-        val arrayList = convertDataToArrayList(resultSet)
-        return arrayList
+            .map { it ->
+                CalibrateResultRow(
+                    it.getString("indicator")!!,
```

Comment:
Shouldn't you be able to use something like `it[ResultData.indicator]` here and avoid the unsafe call and magic strings?